Update error messages to reflect why test is skipped

mikekgfb opened the pull request:
Summary: Update error messages to reflect why test is skipped

Test Plan: github

Differential Revision: D43386390

nmacchioni commented on test/test_transformers.py:
```diff
@@ -1056,7 +1056,7 @@ def ones_tensor(*shape):
                         torch.cuda.synchronize()
 
-    @unittest.skipIf(not PLATFORM_SUPPORTS_FUSED_SDPA or not SM80OrLater, "CUDA unavailable")
+    @unittest.skipIf(not PLATFORM_SUPPORTS_FUSED_SDPA or not SM80OrLater, "platform does not supposrt SDPA or pre-SM80 hardware")
```
*support

nmacchioni commented:
Can we make the messages consistent?

i.e. 
`@unittest.skipIf(not PLATFORM_SUPPORTS_FUSED_SDPA or not SM80OrLater` => `platform does not support fused sdpa or <SM80`,
`@unittest.skipIf(not PLATFORM_SUPPORTS_FUSED_SDPA or not SM86OrLater` => `platform does not support fused sdpa or <SM86`,
`@unittest.skipIf(not PLATFORM_SUPPORTS_FUSED_SDPA` => `platform does not support fused sdpa`,

Commit:
Update error messages to reflect why test is skipped (#95049)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/95049

Update error messages to reflect why test is skipped

Test Plan: github

Reviewed By: cpuhrsch, nmacchioni

Differential Revision: D43386390

fbshipit-source-id: 7e7ca15a3bb14c1c81cdeeb9baee2ec7250237e4

mikekgfb commented:
@pytorchbot merge -f "commit improved skip messages"

seemethere commented:
Heads up, I'm force reverting this since it appears as though we can't revert using the normal tooling

seemethere commented:
Reverted here: https://github.com/pytorch/pytorch/commit/e44737e61975c49b20fa176e77c012b626c5f331

seemethere commented:
@mikekgfb Let's not skip lint failures on these types of PRs in the future. The job should be high signal and should help us avoid issues like this in the future.
